pkg/web/transactions.go

```diff
@@ -311,16 +314,46 @@ func (s *Server) AcceptTransactionPreview(c *gin.Context) {
 		return
 	}
 
-	if out, err = api.NewTransaction(transaction); err != nil {
+	// Retrieve the latest secure envelope from the database
+	if env, err = s.store.LatestSecureEnvelope(ctx, transactionID, models.DirectionAny); err != nil {
+		if errors.Is(err, dberr.ErrNotFound) {
+			c.JSON(http.StatusNotFound, api.Error("latest secure envelope not found"))
+			return
+		}
+
 		c.Error(err)
 		c.JSON(http.StatusInternalServerError, api.Error(err))
 		return
 	}
 
+	// Decrypt the secure envelope using the private keys in the key store
+	if decrypted, err = s.Decrypt(env); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusBadRequest, api.Error("this payload cannot be decrypted"))
+		return
+	}
+
+	if out, err = api.NewEnvelope(decrypted); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusInternalServerError, api.Error(err))
+		return
+	}
+
+	// Create the web response, which also includes the transaction details.
+	webData := gin.H{
+		"envelope":    out,
+		"transaction": nil,
+	}
+
+	if webData["transaction"], err = api.NewTransaction(transaction); err != nil {
+		c.Error(err)
+	}
```

Comment:
Here's the real question -- the API just gets the secure envelope, but I'm also returning the `*models.Transaction` for the webdata. However, if you don't need this, I'd rather exclude it. The envelope has a `Transaction` field (or a `Pending` field) with the data about the chain, originator and beneficiary crypto addresses, amount, etc. The EnvelopeID is the same as the TransactionID. The information you'd be losing would be `source`, `status`, `counterparty`, `counterparty_id`, and `envelope_count`. Any thoughts?

Reply:
We have the ability to check the source to determine if `accept` or `reject` should be displayed as an action in the transaction list, so I think that not returning the `*models.Transaction` here will be fine. Other info like `status`, `envelope_count`, etc. is still provided to `transaction_detail.html` so users may view it on the detail page or the list page.

Reply:
Ok, in that case I'm going to remove it -- which will definitely simplify things!